dunstctl

```diff
@@ -8,7 +8,12 @@ DBUS_IFAC_DUNST="org.dunstproject.cmd0"
 DBUS_IFAC_PROP="org.freedesktop.DBus.Properties"
 DBUS_IFAC_FDN="org.freedesktop.Notifications"
 
-die(){ printf "%s\n" "${1}" >&2; exit 1; }
+die() {
+	printf "%s\n" "${1}" >&2
+	[ -z "$DBUS_SESSION_BUS_ADDRESS" ] \
+		&& echo "DBUS_SESSION_BUS_ADDRESS is blank. Is the dbus session configured correctly?"
```

Comment:
You're having `-e` set int the shell. Testing something for negative result and then concatenating the resulting with `&&` will make the shell totally abort without even printing an error message.

```suggestion
	if [ -z "$DBUS_SESSION_BUS_ADDRESS" ]; then
		echo "DBUS_SESSION_BUS_ADDRESS is blank. Is the D-Bus session configured correctly?" >&2
	fi
```

Reply:
Alternatively you can use `||`:

```suggestion
	[ -n "$DBUS_SESSION_BUS_ADDRESS" ] \
		|| echo "DBUS_SESSION_BUS_ADDRESS is blank. Is the dbus session configured correctly?"
```

But I've learnt using an `if` is much better, because nobody sees visibly this intricate behavioral difference.

Reply:
thanks